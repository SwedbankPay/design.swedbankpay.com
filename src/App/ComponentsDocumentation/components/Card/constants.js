import React from "react";
import CardsComponent from "@components/Cards";

const BASENAME = process.env.basename;

export const overviewCards = {
    id: "overviewCards",
    tabsId: "overviewButtonsTabs",
    elements: [
        {
            tab: "Standard",
            component: <CardsComponent
                titleTxt="Title"
                type="primary"
                text="Write the supportive paragraph here."
                icon={<i className="material-icons">shopping_basket</i>}
            />,
            options: {
                radio: [
                    {
                        id: "addons_radio",
                        title: "Addons",
                        values: [
                            {
                                name: "Plain",
                                value: {
                                    icon: null
                                }
                            },
                            {
                                name: "With icon",
                                default: true,
                                value: {
                                    icon: <i className="material-icons">shopping_basket</i>
                                }
                            },
                            {
                                name: "With image",
                                value: {
                                    imgSrc: `${BASENAME}img/documentation/cards/img-example.png`,
                                    icon: null
                                }
                            }
                        ]
                    },
                    {
                        id: "color_radio",
                        title: "Accent color",
                        values: [
                            {
                                name: "Yellow brown",
                                value: {
                                    type: "primary"
                                }
                            },
                            {
                                name: "Turquoise",
                                value: {
                                    type: "secondary"
                                }
                            },
                            {
                                name: "Purple",
                                value: {
                                    type: "tertiary"
                                }
                            }
                        ]
                    }
                ]
            },
            title: "Standard cards",
            description: "The standard card component can be customized either icon or picture in order to make it easier for users to scan the content. Mixing simple card without icon should mostly be done when seperating the cards functionality such as when there are more cards to be displayed or when approperiate icons can not be found."
        },
        {
            tab: "Wide",
            component: <CardsComponent
                titleTxt="Title"
                type="primary"
                text="Write the supportive paragraph here."
                wide
                icon={<>01</>}
            />,
            options: {
                radio: [
                    {
                        id: "addons_radio",
                        title: "Addons",
                        values: [
                            {
                                name: "Plain",
                                value: {
                                    icon: null
                                }
                            },
                            {
                                name: "With icon",
                                default: true,
                                value: {
                                    icon: <>01</>
                                }
                            }
                        ]
                    },
                    {
                        id: "color_radio",
                        title: "Accent color",
                        values: [
                            {
                                name: "Yellow brown",
                                value: {
                                    type: "primary"
                                }
                            },
                            {
                                name: "Turquoise",
                                value: {
                                    type: "secondary"
                                }
                            },
                            {
                                name: "Purple",
                                value: {
                                    type: "tertiary"
                                }
                            }
                        ]
                    }
                ]
            },
            title: "Wide cards",
            description: "The wide card component can be used with and without numbering. Try to avoid mixing different card setups with each other as it could be confusing for the users to understand. Try to keep the title concise and to the point, to make it simple to understand and the design aligned with the cards as a group."
        }
    ]
};
