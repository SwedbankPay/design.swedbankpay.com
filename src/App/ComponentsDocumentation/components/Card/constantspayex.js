import React from "react";
import CardsComponent from "@components/Cards";

const defaultOptionsValues = {
	isButton: false,
	isWide: false,
	iconClasses: "at-document",
	hasTitle: true,
	imgSrc: null,
	imgRatio: null,
	hasTextContent: true,
	hasCTAText: true,
};

// Happy Woman image

const brand = process.env.brand;

const imgSrcURL =
	brand === "payex"
		? "https://design.payex.com/v/10.10.1/img/payex/documentation/cards/happy-woman.png"
		: "https://design.swedbankpay.com/v/10.10.1/img/documentation/imagery/women-on-bus.svg";

export const overviewCards = {
	id: "overviewCards",
	tabsId: "overviewButtonsTabs",
	elements: [
		{
			tab: "Standard",
			component: (
				<CardsComponent
					titleTxt="Title"
					textContent="Write the supportive paragraph here."
					hasCTAText={defaultOptionsValues.hasCTAText}
					isButton={defaultOptionsValues.isButton}
					isWide={false}
					hasTitle={defaultOptionsValues.hasTitle}
					imgSrc={defaultOptionsValues.imgSrc}
					iconClasses={defaultOptionsValues.iconClasses}
					hasTextContent={defaultOptionsValues.hasTextContent}
					imgRatio={defaultOptionsValues.imgRatio}
				/>
			),
			options: {
				radio: [
					{
						id: "title",
						title: "Title",
						values: [
							{
								name: "Text only",
								value: {
									iconClasses: null,
									hasTitle: true,
								},
							},
							{
								name: "Text + icon",
								default: true,
								value: {
									iconClasses: "at-shop-cart",
									hasTitle: true,
								},
							},
							{
								name: "Icon only",
								value: {
									iconClasses: "at-shop-cart",
									hasTitle: false,
								},
							},
						],
					},
					{
						id: "has_image",
						title: "Image illustration",
						values: [
							{
								name: "No image",
								default: true,
								value: {
									imgSrc: null,
								},
							},
							{
								name: "With image - 16/9 ratio",
								value: {
									imgSrc:
										"https://design.swedbankpay.com/v/10.10.1/img/documentation/imagery/women-on-bus.svg",
									icon: null,
									imgRatio: "ratio-16-9",
								},
							},
							{
								name: "With image - 4/3 ratio",
								value: {
									imgSrc:
										"https://design.swedbankpay.com/v/10.10.1/img/documentation/imagery/women-on-bus.svg",
									icon: null,
									imgRatio: "ratio-4-3",
								},
							},
							{
								name: "With image - 1/1 ratio",
								value: {
									imgSrc:
										"https://design.swedbankpay.com/v/10.10.1/img/documentation/imagery/women-on-bus.svg",
									icon: null,
									imgRatio: "ratio-1-1",
								},
							},
						],
					},
					{
						id: "text_content",
						title: "Text content",
						values: [
							{
								name: "With text content",
								default: true,
								value: {
									hasTextContent: true,
								},
							},
							{
								name: "No text content",
								value: {
									hasTextContent: false,
								},
							},
						],
					},
					{
						id: "has_cta_text",
						title: "CTA text",
						values: [
							{
								name: "With CTA text",
								default: true,
								value: {
									hasCTAText: true,
								},
							},
							{
								name: "Only arrow",
								value: {
									hasCTAText: false,
								},
							},
						],
					},
					{
						id: "card_element_tag",
						title: "Card purpose",
						values: [
							{
								name: "action <button>",
								value: {
									isButton: true,
								},
							},
							{
								name: "navigation link <a>",
								default: true,
								value: {
									isButton: false,
								},
							},
						],
					},
				],
			},
			title: "Standard cards",
			description:
				"The standard card component can be customized either icon or picture in order to make it easier for users to scan the content. Mixing simple card without icon should mostly be done when seperating the cards functionality such as when there are more cards to be displayed or when approperiate icons can not be found.",
		},

		{
			tab: "Wide",
			component: (
				<CardsComponent
					titleTxt="Title"
					textContent="Write the supportive paragraph here."
					hasCTAText={defaultOptionsValues.hasCTAText}
					isButton={defaultOptionsValues.isButton}
					isWide={true}
					hasTitle={defaultOptionsValues.hasTitle}
					imgSrc={defaultOptionsValues.imgSrc}
					iconClasses={defaultOptionsValues.iconClasses}
					hasTextContent={defaultOptionsValues.hasTextContent}
					imgRatio={defaultOptionsValues.imgRatio}
				/>
			),
			options: {
				radio: [
					{
						id: "title",
						title: "Title",
						values: [
							{
								name: "Text only",
								value: {
									iconClasses: null,
									hasTitle: true,
								},
							},
							{
								name: "Text + icon",
								default: true,
								value: {
									iconClasses: "at-shop-cart",
									hasTitle: true,
								},
							},
							{
								name: "Icon only",
								value: {
									iconClasses: "at-shop-cart",
									hasTitle: false,
									hasCTAText: true,
								},
							},
						],
					},
					{
						id: "has_image",
						title: "Image illustration",
						values: [
							{
								name: "No image",
								default: true,
								value: {
									imgSrc: null,
								},
							},
							{
								name: "With image",
								value: {
									imgSrc:
										"https://design.swedbankpay.com/v/10.10.1/img/documentation/imagery/women-on-bus.svg",
									icon: null,
									imgRatio: null,
								},
							},
						],
					},
					{
						id: "text_content",
						title: "Text content",
						values: [
							{
								name: "With text content",
								default: true,
								value: {
									hasTextContent: true,
								},
							},
							{
								name: "No text content",
								value: {
									hasTextContent: false,
								},
							},
						],
					},
					{
						id: "has_cta_text",
						title: "CTA text",
						values: [
							{
								name: "With CTA text",
								default: true,
								value: {
									hasCTAText: true,
								},
							},
							{
								name: "Only arrow",
								value: {
									hasCTAText: false,
								},
							},
						],
					},
					{
						id: "card_element_tag",
						title: "Card purpose",
						values: [
							{
								name: "action <button>",
								value: {
									isButton: true,
								},
							},
							{
								name: "navigation link <a>",
								default: true,
								value: {
									isButton: false,
								},
							},
						],
					},
				],
			},
			title: "Wide cards",
			description:
				"The wide card component can be customized either icon or picture in order to make it easier for users to scan the content. Try to avoid mixing different card setups with each other as it could be confusing for the users to understand. Try to keep the title concise and to the point, to make it simple to understand and the design aligned with the cards as a group.",
		},
	],
};