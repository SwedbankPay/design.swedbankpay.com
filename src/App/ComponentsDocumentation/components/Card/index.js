import React, { Component } from "react";
import { Link } from "react-router-dom";

import { ComponentPreview, DocContainer } from "@docutils";
import CardComponent from "@components/Card";
import MediaObject from "@components/MediaObject";
import CodeTags from "@components/CodeTags";

const textArr = ["This is a lot of text", "With some more text", "And then even some more", "Is it really possible to have this much text in one card?", "Yes!"];

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>
            Cards support a wide variety of content, including images, text, list groups, links, and more.
            Make sure to restrain the width of your card unless you wish for it to fill the wrapping component.
        </p>
        <ComponentPreview language="html" showCasePanel showCasePanelSm codeFigure>
            <CardComponent
                titleTxt="Card title"
                type="primary"
                imgSrc="https://picsum.photos/300/?random"
                text="This paragraph contains some text related to this card."
                footerTxt="Footer text"
            />
        </ComponentPreview>
    </>
);

const GridCard = () => (
    <>
        <h2 id="card-with-grid">Card with grid</h2>
        <p>
            Use our <Link to="/docs/core/grid">grid</Link> along with cards to control their size and how they are displayed on different screen resolutions.
            Make sure to add <CodeTags type="secondary" code=".d-flex" /> to the wrapper if you want the cards to be of equal height when next to each other.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="row">
                <div className="col-12 col-lg-6 col-xl-4 d-flex">
                    <CardComponent
                        type="primary"
                        titleTxt="My grid card"
                        text="Card text"
                        smallText="Small card text"
                        btn
                        btnClass="btn-block mt-auto"
                        bodyClass="d-flex flex-column"
                    >
                        <MediaObject size="sm" icon="verified_user" heading="75%" text="Complete" textSmall muted />
                    </CardComponent>
                </div>

                <div className="col-12 col-lg-6 col-xl-4 d-flex">
                    <CardComponent
                        type="primary"
                        titleTxt="My grid card"
                        textSection={textArr}
                        smallText="This is some small text"
                        btn
                        btnClass="btn-block mt-auto"
                        bodyClass="d-flex flex-column"
                    />
                </div>

                <div className="col-12 col-lg-6 col-xl-4 d-flex">
                    <CardComponent
                        type="primary"
                        titleTxt="My grid card"
                        imgSrc="https://picsum.photos/300/?random"
                        text="This is a card with an image and body"
                        footerLinkTxt="Footer link"
                    />
                </div>
            </div>
        </ComponentPreview>
    </>
);

const CardTypes = () => (
    <>
        <h2 id="card-types">Card Types</h2>
        <p>
            We offer three different card types; <CodeTags type="secondary" code=".card-primary" />, <CodeTags type="secondary" code=".card-secondary" /> and <CodeTags type="secondary" code=".card-plain" />.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="row">
                <div className="col-12 col-lg-6 col-xl-4 d-flex">
                    <CardComponent
                        type="primary"
                        titleTxt="Card Primary"
                        imgSrc="https://picsum.photos/300/?random"
                        text="This paragraph contains some text about the person displayed above."
                    >
                        <ul className="list">
                            <li><span className="font-weight-bold">Card number:</span> 4563 5648 5642</li>
                            <li><span className="font-weight-bold">Balance:</span> 2500,- </li>
                        </ul>
                    </CardComponent>
                </div>
                <div className="col-12 col-lg-6 col-xl-4 d-flex">
                    <CardComponent
                        type="secondary"
                        titleTxt="Card Secondary"
                        text="Descriptive text to put inside the card, could contain a lot."
                        footerLinkTxt="Footer link"
                    >
                        <MediaObject size="sm" icon="account_balance" heading="725 NOK" text="Outstanding balance" textSmall muted />
                    </CardComponent>
                </div>
                <div className="col-12 col-lg-6 col-xl-4 d-flex">
                    <CardComponent
                        type="plain"
                        titleTxt="Card Plain"
                        imgSrc="https://picsum.photos/300/?random"
                        text="Descriptive text to put inside the card, could contain a lot."
                        footerLinkTxt="Footer link"
                    >
                        <MediaObject size="sm" icon="verified_user" heading="75%" text="Complete" textSmall muted />
                    </CardComponent>
                </div>
            </div>
        </ComponentPreview>
    </>
);

class Card extends Component {
    render () {
        return (
            <DocContainer docToc>
                <p className="lead">
                    Card is a flexible colored box which fills the width of its parent element.
                </p>
                <Overview />
                <GridCard />
                <CardTypes />
            </DocContainer>
        );
    }
}

export default Card;

/* For testing */
export { Overview, GridCard };
