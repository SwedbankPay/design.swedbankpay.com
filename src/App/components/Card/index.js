import React from "react";
import PropTypes from "prop-types";

const Card = ({ type, titleTxt, titleMarkup, imgSrc, text, textSection, smallText, btn, btnClass, btnText, bodyClass, footerTxt, footerLink, footerLinkTxt, children }) => (
    <div className={`card${type ? ` card-${type}` : ""}`}>
        {imgSrc ? <div className="card-img">{"\n"}
            <img src={imgSrc} alt=""/>{"\n"}
        </div> : null}
        {titleTxt
            ? (<header className="card-header">
                <h3>{titleTxt}</h3>
            </header>)
            : (titleMarkup
                ? (<header className="card-header">
                    {titleMarkup}
                </header>)
                : null)}
        <div className={`card-body${bodyClass ? ` ${bodyClass}` : ""}`}>
            {children ? children : null }
            {text ? <p>{text}</p> : null}{"\n"}
            {textSection ?
                textSection.map((secText, i) => <p key={`card-p-${i}`}>{secText}</p>)
                : null
            }
            {btn ? <button type="button" className={`btn btn-executive${btnClass ? ` ${btnClass}` : ""}`}>{"\n"}{btnText}{"\n"}</button> : null}
        </div>
        {footerTxt || footerLinkTxt
            ? <footer className="card-footer">
                {smallText ? <small>{smallText}</small> : null}
                {footerTxt ? footerTxt : null}
                {footerLinkTxt ? <div className="footer-link"><a href={footerLink || "#"}>{footerLinkTxt}</a></div> : null}
            </footer> : null}
    </div>
);

Card.propTypes = {
    type: PropTypes.oneOf(["primary", "secondary", "plain"]),
    imgSrc: PropTypes.string,
    titleTxt: PropTypes.string,
    titleMarkup: PropTypes.object,
    icon: PropTypes.string,
    text: PropTypes.string,
    smallText: PropTypes.string,
    textSection: PropTypes.array,
    btn: PropTypes.bool,
    btnText: PropTypes.string,
    footerLink: PropTypes.string,
    footerLinkTxt: PropTypes.string
};

export default Card;
