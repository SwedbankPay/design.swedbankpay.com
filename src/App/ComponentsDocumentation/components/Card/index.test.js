import React from "react";
import { shallow } from "enzyme";

import Card, { Overview, GridCard } from "./index";

describe("Documentation: Card", () => {
    it("is defined", () => {
        expect(Card).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Card />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("Overview", () => {
        it("is defined", () => {
            expect(Overview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Overview />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("GridCard", () => {
        it("is defined", () => {
            expect(GridCard).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<GridCard />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
