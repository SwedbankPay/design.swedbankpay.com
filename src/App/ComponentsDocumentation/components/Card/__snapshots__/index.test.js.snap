// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation: Card GridCard renders 1`] = `
<Fragment>
  <h2
    id="card-with-grid"
  >
    Card with grid
  </h2>
  <p>
    Use our 
    <Link
      to="/docs/core/grid"
    >
      grid
    </Link>
     along with cards to control their size and how they are displayed on different screen resolutions. Make sure to add 
    <CodeTags
      code=".d-flex"
      type="secondary"
    />
     to the wrapper if you want the cards to be of equal height when next to each other.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="row"
    >
      <div
        className="col-12 col-lg-6 col-xl-4 d-flex"
      >
        <Card
          bodyClass="d-flex flex-column"
          btn={true}
          btnClass="btn-block mt-auto"
          smallText="Small card text"
          text="Card text"
          titleTxt="My grid card"
          type="primary"
        >
          <MediaObject
            heading="75%"
            icon="verified_user"
            muted={true}
            size="sm"
            text="Complete"
            textSmall={true}
          />
        </Card>
      </div>
      <div
        className="col-12 col-lg-6 col-xl-4 d-flex"
      >
        <Card
          bodyClass="d-flex flex-column"
          btn={true}
          btnClass="btn-block mt-auto"
          smallText="This is some small text"
          textSection={
            Array [
              "This is a lot of text",
              "With some more text",
              "And then even some more",
              "Is it really possible to have this much text in one card?",
              "Yes!",
            ]
          }
          titleTxt="My grid card"
          type="primary"
        />
      </div>
      <div
        className="col-12 col-lg-6 col-xl-4 d-flex"
      >
        <Card
          footerLinkTxt="Footer link"
          imgSrc="https://picsum.photos/300/?random"
          text="This is a card with an image and body"
          titleTxt="My grid card"
          type="primary"
        />
      </div>
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Card Overview renders 1`] = `
<Fragment>
  <h2
    id="overview"
  >
    Overview
  </h2>
  <p>
    Cards support a wide variety of content, including images, text, list groups, links, and more. Make sure to restrain the width of your card unless you wish for it to fill the wrapping component.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
    showCasePanelSm={true}
  >
    <Card
      footerTxt="Footer text"
      imgSrc="https://picsum.photos/300/?random"
      text="This paragraph contains some text related to this card."
      titleTxt="Card title"
      type="primary"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Card renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Card is a flexible colored box which fills the width of its parent element.
  </p>
  <Overview />
  <GridCard />
  <CardTypes />
</DocContainer>
`;
